### ISSUE TITLE
tween() without start parameter non-linear fade

### ISSUE THREAD

**Issue by rowlap** [pimoroni/piglow#22]
`start` is an optional parameter to `tween()`, with the documented default of "the current state".

The problem is the implementation literally assigns a reference to the `_values` list, without taking a copy. As such, when `_values` changes during the loop, so does `start` as they refer to the same list.

Link to [code at time of writing](https://github.com/pimoroni/piglow/blob/63b4f16b7b2cb16033994c305e5d1d336559a7f6/library/piglow.py#L233-L263).

Steps to reproduce:
```
from piglow import tween

for _ in range(3):
    tween(2.0, [255] * 18)
    tween(2.0, [0] * 18)

for _ in range(3):
    tween(2.0, start=[0]*18, end=[255]*18)
    tween(2.0, start=[255]*18, end=[0]*18)
```

What happened:
In the first loop, LEDs fade in/out much faster than the second loop.

What's expected:
Both loops give the same result.

**Comment by Gadgetoid** [comment]
I wish every issue was this detailed and descriptive. Thank you for reporting and subsequently fixing this.

**Comment by Gadgetoid** [comment]
Published in release v1.2.5 - https://github.com/pimoroni/piglow/releases/tag/v1.2.5
